flink-core/src/main/java/org/apache/flink/api/common/serialization/SerializerConfig.java

```diff
@@ -229,8 +233,9 @@ public LinkedHashSet<Class<?>> getRegisteredPojoTypes() {
     }
 
     /** Returns the registered type info factories. */
-    public LinkedHashMap<Class<?>, Class<TypeInfoFactory<?>>> getRegisteredTypeFactories() {
-        return registeredTypeFactories;
+    public LinkedHashMap<Class<?>, Class<? extends TypeInfoFactory<?>>>
```

Comment:
The type should be `Class<? extends TypeInfoFactory<?>>` instead of `Class<TypeInfoFactory<?>>` for proper functioning. Updated in the FLIP as well